rust/src/asn1/parse_rules.rs

```diff
@@ -74,7 +74,7 @@ pub unsafe extern "C" fn rs_detect_asn1_parse(input: *const c_char) -> *mut Dete
 ///
 /// ptr must be a valid object obtained using `rs_detect_asn1_parse`
 #[no_mangle]
-pub unsafe extern "C" fn rs_detect_asn1_free(ptr: *mut DetectAsn1Data) {
```

Comment:
is this a formatting change?

Reply:
Yes.  By default rustfmt will convert all `extern {` (without a language) to `extern "C"` just to be explicit by default.  This causes issues in an extern block that creates bindings to C functions, for example:
```
extern "C" {
    pub fn AppLayerRegisterProtocolDetection(
        parser: *const RustParser, enable_default: c_int,
    ) -> AppProto;
    pub fn AppLayerRegisterParserAlias(parser_name: *const c_char, alias_name: *const c_char);
}
```
When the `"C"` is there, `cbindgen` will go and re-export these methods in the generated header which our C code doesn't like.  And `cbindgen` has no option to turn them off.  This does not happen when the "C" is left out, and "C" is the default when left out.

This is why I had to add the following to `rustfmt.toml`:
```
force_explicit_abi = false
```

The other options is to explicitly ignore every function we don't want re-exported in the cbindgen configuration, but thats a little tedious.

Reply:
Not sure I fully get it. Did rustfmt make these changes despite `force_explicit_abi = false` or are you making that change in a follow up PR?

Reply:
(in general a reformatting tool doing things that have effects beyond formatting is ... ugh. It matters because it means I now distrust the tool in terms of not producing side effects)

Reply:
Rust format made these changes.  I've instructed it to just use `extern {` and not `extern "C" {`. This is side affect free to Rust and purely a matter of style.  An `extern "WASM"` would not be changed.

This is just a matter of consistency.  I'd rather be consistent on `extern "C" {`, but cbindgen is broken a bit here, which is another reason to be consistent about it as well.

This is a change I'd like to make even without going ahead with rustfmt.